### PR TITLE
Update Pymc3_tips_and_heuristics notebook

### PR DESCRIPTION
**Pull request by AmitKus**
1) Fixed KeyError: Rhat
2) Fixed: FutureWarnings
3) Updated the notebook to follow the consistent style (https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style)
4) Use arviz functions directly for plotting the inference data

**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [ ] consider adding/updating relevant example notebooks
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Comment by fonnesbeck** [comment]
Since this notebook is being updated, it should probably be renamed. Its not a general notebook about PyMC3 tips, as the name implies; Its a very specific case study of CAR modeling.

**Comment by AmitKus** [comment]
> Since this notebook is being updated, it should probably be renamed. Its not a general notebook about PyMC3 tips, as the name implies; Its a very specific case study of CAR modeling.

How about simply renaming it to conditional-autoregressive-model.ipynb?

**Comment by AmitKus** [comment]
<p>Without ADVI, it takes 20X longer for the same settings and some chains contain only diverging samples.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc3/pull/4036/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AmitKus** [comment]
<p>Similar story without ADVI here too. Sampling takes an order of magnitude longer.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc3/pull/4036/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AmitKus** [comment]
<pre class="ql-syntax" spellcheck="false">Sampling 4 chains for 500 tune and 1_000 draw iterations (2_000 + 4_000 draws total) took 3015 seconds.
There were 41 divergences after tuning. Increase <code>target_accept</code> or reparameterize.
The acceptance probability does not match the target. It is 0.7040791687604867, but should be close to 0.9. Try to increase the number of tuning steps.
There were 993 divergences after tuning. Increase <code>target_accept</code> or reparameterize.
The acceptance probability does not match the target. It is 7.239174982000396e-38, but should be close to 0.9. Try to increase the number of tuning steps.
There were 4 divergences after tuning. Increase <code>target_accept</code> or reparameterize.
The acceptance probability does not match the target. It is 0.6825918614079135, but should be close to 0.9. Try to increase the number of tuning steps.
The acceptance probability does not match the target. It is 0.7627370968011182, but should be close to 0.9. Try to increase the number of tuning steps.
The rhat statistic is larger than 1.4 for some parameters. The sampler did not converge.
The estimated number of effective samples is smaller than 200 for some parameters.
</pre><p> </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc3/pull/4036/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AmitKus** [comment]
<pre class="ql-syntax" spellcheck="false">Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (4 chains in 4 jobs)
NUTS: [mu_phi, theta, tau_c, tau_h, beta1, beta0]
</pre><p>&nbsp;100.00% [6000/6000 1:06:22&lt;00:00 Sampling 4 chains, 1,001 divergences]</p><pre class="ql-syntax" spellcheck="false">Sampling 4 chains for 500 tune and 1_000 draw iterations (2_000 + 4_000 draws total) took 3984 seconds.
The chain contains only diverging samples. The model is probably misspecified.
The acceptance probability does not match the target. It is 0.0037493742425586075, but should be close to 0.9. Try to increase the number of tuning steps.
There was 1 divergence after tuning. Increase <code>target_accept</code> or reparameterize.
The acceptance probability does not match the target. It is 0.7159791144918716, but should be close to 0.9. Try to increase the number of tuning steps.
The rhat statistic is larger than 1.4 for some parameters. The sampler did not converge.
The estimated number of effective samples is smaller than 200 for some parameters.
</pre><p> </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc3/pull/4036/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by AmitKus** [comment]
<p>Fixed this and all the other suggestions above. Thanks.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc3/pull/4036/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by AlexAndorra — APPROVED** [review]
Looks all good now, thanks @AmitKus !
Good idea: let's just rename it `conditional-autoregressive-model` 👍 
Also, don't forget to run black_nbconvert`, as per the [style guide](https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style) 😉

**Comment by junpenglao** [comment]
I remember last time when I was updating the notebook, inference without ADVI is pretty problematic - I think you should try more informative prior.

Also, since you are renaming the file, you should do a grep and rename other places where we reference the name (mostly in the doc set up)

File: docs/source/notebooks/table_of_contents_tutorials.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gallery.contents = {
     "ODE_API_shapes_and_benchmarking": "Deep dives",
     "DEMetropolisZ_EfficiencyComparison": "Deep dives",
     "DEMetropolisZ_tune_drop_fraction": "Deep dives",
-    "PyMC3_tips_and_heuristic": "How-To",
+    "conditional-autoregressive-model": "How-To",
```
Comment (junpenglao) [inline docs/source/notebooks/table_of_contents_tutorials.js]: Could you move this from "How-to" to Gaussian Process?

Comment (AmitKus) [reply docs/source/notebooks/table_of_contents_tutorials.js]: done

**Review by AlexAndorra — CHANGES_REQUESTED** [review]
Thanks @AmitKus, looking good now! Last necessary changes I think:
- Did you look for other places where the old NB name was used and replaced it with the new name, as @junpenglao suggested? The references will probably be concentrated in the docs set-up.
- I think the cell with all the data is overcrowded and should be replaced (see my comment below).
- Don't forget to run `black_nbconvert` when the NB is ready, as explained in the [style guide](https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style).

**Comment by AmitKus** [comment]
> Thanks @AmitKus, looking good now! Last necessary changes I think:
> 
>     * Did you look for other places where the old NB name was used and replaced it with the new name, as @junpenglao suggested? The references will probably be concentrated in the docs set-up.
> 
>     * I think the cell with all the data is overcrowded and should be replaced (see my comment below).
> 
>     * Don't forget to run `black_nbconvert` when the NB is ready, as explained in the [style guide](https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style).

1. I could only find the old name at one location which I already changed.
2. Moved the data to csv file which I have placed in pymc3/examples/data folder.
3. Ran the formatting.
Please let me know if something is still missing.
Thanks,
Amit

**Review by AlexAndorra — APPROVED** [review]
This is all good now, thanks a lot @AmitKus !

**Pull request merged**